src/main/java/org/folio/edge/sip2/handlers/freemarker/FreemarkerUtils.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.edge.sip2.handlers.freemarker;
+
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.lang.invoke.MethodHandles;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class FreemarkerUtils {
+
+  private static Logger log;
+
+  static {
+    log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+  }
+
+  /**
+   * Utility method that executes a freemarker template to return a templated-formmated string.
+   *
+   * @param data the data used by the template.
+   * @param template the template to apply to the data.
+   * @return String output of running the template on the data
+   */
+  public static String executeFreemarkerTemplate(Object data, Template template) {
+
+    Writer out = new StringWriter();
+    String outputString = "";
+
+    try {
+      template.process(data, out);
+      outputString = out.toString();
+    } catch (TemplateException e) {
+      log.error("Having problems finding and loading template: " + e.getMessage());
+    } catch (IOException ioEx) {
+      log.error("Having problems applying template to data: " + ioEx.getMessage());
+    } catch (Exception ex) {
+      log.error("Error applying template to data: " + ex.getMessage());
+    }
+
+    log.debug("Data =" + (data == null ? "" : data.toString())
+              + "Template =" + (template == null ? "" : template.getName()));
```

Comment:
We might want to build this via the logger, so we don't construct a string that will get thrown away when logging is not enabled,  something like:
```suggestion
    log.debug("Data = {} Template = {}",  data == null ? "" : data.toString(), template == null ? "" : template.getName());
```
Or better yet:
```suggestion
    log.debug("Data = {} Template = {}", () -> {data == null ? "" : data.toString()}, () -> {template == null ? "" : template.getName()});
```
With this approach, `data.toString()`s and the `getName()` are not called unless debug is executed.

Reply:
debug()'s signature does not accept lambdas.

Reply:
I believe it is supported via the `debug(String message, Supplier<?>... paramSuppliers`.

[Example](https://logging.apache.org/log4j/2.0/manual/api.html#Java_8_lambda_support_for_lazy_logging)

Reply:
The braces after ()-> threw it off. I removed the braces and it's working now.

Reply:
Sorry, I thought they might be needed since these were ternary operations, so I added them. I guess I was wrong. 😄